src/main/java/com/puppycrawl/tools/checkstyle/utils/AnnotationUtil.java

```diff
@@ -86,10 +84,47 @@ public static boolean containsAnnotation(final DetailAST ast) {
         return holder != null && holder.findFirstToken(TokenTypes.ANNOTATION) != null;
     }
 
+    /**
+     * Checks if the given AST element is annotated with any of the specified annotations.
+     *
+     * <p>
+     * This method accepts both simple and fully-qualified names,
+     * e.g. "Override" will match both java.lang.Override and Override.
+     * </p>
+     *
+     * @param ast The type or method definition.
+     * @param annotations A collection of annotations to look for.
+     * @return {@code true} if the given AST element is annotated with
+     *                      at least one of the specified annotations;
+     *                      {@code false} otherwise.
+     */
+    public static boolean containsAnnotation(DetailAST ast, List<String> annotations) {
+        if (ast == null) {
+            throw new IllegalArgumentException(THE_AST_IS_NULL);
+        }
+
+        boolean result = false;
+
+        if (annotations != null && !annotations.isEmpty()) {
+            final DetailAST firstMatchingAnnotation = getAnnotation(ast, annotationNode -> {
```

Comment:
`firstMatchingAnnotation` is a better name because it tells the developer who'll read the code that if there are multiple annotations in the `annotations` list, only the first one will be returned.

Reply:
Ok, please rename `getAnnotation` to mention that you return only first found item.

Reply:
It was already renamed to `findAnnotation` as per your earlier comment, I will rename it to `findFirstAnnotation`